platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SignedStateFileManager.java

```diff
@@ -332,6 +336,11 @@ private boolean shouldSaveToDisk(final SignedState signedState, final Instant pr
             return false;
         }
 
+        if (source == SourceOfSignedState.RECONNECT && stateConfig.saveReconnectStateToDisk()) {
+            // states received via reconnect should be written to disk if configured
+            return true;
+        }
+
         if (previousTimestamp == null) {
```

Comment:
The argument on the method says NonNull, but there is a null test here in the code.  Is the argument `@Nullable`?

Reply:
good catch, fixed